### ISSUE TITLE
WikiText2 dataset incorrect targets

### ISSUE THREAD

**Issue by A-Jacobson** [pytorch/text#207]
With torchtext v 0.2.1, It seems that the targets returned by the WikiText2 dataset correspond to the next window of `bptt_len` over the dataset. For a language modeling task, shouldn't the targets be simply the next word? 

concretely: `batch.text =  ['<eos>', '=', 'Valkyria', 'Chronicles', 'III']` and `batch.target =  ['=', 'Valkyria', 'Chronicles', 'III', '=']` and it is my understanding that `batch.target = ['=']` would be correct.

full testing code:
```python

from torchtext import data
from torchtext import datasets

TEXT = data.Field()

train_data, val_data, test_data = datasets.WikiText2.splits(TEXT)

TEXT.build_vocab(train_data, min_freq=10)

train_iter, val_iter, test_iter = data.BPTTIterator.splits(
            (train_data, val_data, test_data),
            batch_size=1, bptt_len=5, repeat=False)

batch = next(iter(train_iter))

def recover_text(vector):
    return [TEXT.vocab.itos[int(v)] for v in vector]

recover_text(batch.text.data.squeeze())  # >>> ['<eos>', '=', 'Valkyria', 'Chronicles', 'III']

recover_text(batch.target.data.squeeze()) # >>> ['=', 'Valkyria', 'Chronicles', 'III', '=']
```

Thanks for the awesome package.

-Austin

**Comment by keon** [comment]
```
  RNN      RNN      RNN           RNN        RNN
'<eos>',  '=',  'Valkyria',  'Chronicles',  'III'
```

**Comment by jekbradbury** [comment]
The targets correspond to the next word at each timestep in the batch, if that makes sense. The LM is typically provided with an entire window of `bptt_len` at a time, and the softmax produces `bptt_len`  next-word predictions, which are supervised by `bptt_len` targets. You can check out the PyTorch word language modeling example or other LM implementations for more info on how that works.

**Comment by A-Jacobson** [comment]
Oh, thanks a lot James, I had only played with kaparthyesque toy models before, but wanted to build a neural cache model . I wasn't aware that was the way targets are meant to be set up as most of the papers only mention predicting the next word. I apologize for the false alarm!
